### PR TITLE
Feature: Add Support for Buildless

### PR DESCRIPTION
**Pull request by sgammon**
## Summary

Draft PR which adds [Buildless][0] support to `sccache`. If this goes well, later, we can maybe upstream it.

### Task list

- [ ] Produce an initial release
- [ ] **Backend Support:** Add Buildless as a supported `sccache` backend
  - [x] Add configuration support for Buildless
  - [x] Implement WebDAV via "generic cache" endpoints
  - [x] Implement RESP (Redis) via Buildless Cloud edge
  - [x] Implement detection of local Buildless Agent
  - [x] Detect and enable with `BUILDLESS_APIKEY`
- [ ] **Docs:** Documenting this integration for both `sccache` and Buildless
  - [x] Add doc to `sccache`
  - [x] Update `sccache` readme
  - [x] Add Buildless docs for `sccache` integration
  - [ ] Add Buildless docs for `sccache` fork and feature
- [ ] **Testing:** Testing new functionality added in this PR
  - [x] Add CI integration tests for `sccache` + `buildless` cloud
  - [ ] Add CI integration tests for `sccache` + `buildless` agent
  - [ ] Add basic configuration unit tests
  - [ ] Add tests for env-variable detection and config
  - [ ] Add tests for local agent + `sccache`
  - [ ] Add tests for cloud edge + `sccache`
- [ ] **Embedded:** Investigate embedding `sccache` in releases of the `buildless` CLI
  - [ ] License compatibility
  - [ ] Binary size / embedding
  - [ ] Invocation from GVM

## Changelog

- feat(buildless): add configuration structures for buildless
- feat(buildless): implement cache backend for buildless, using webdav
- feat(buildless): implement cache backend for buildless, using redis
- feat(buildless): implement agent detection and use (where enabled)
- feat(buildless): decode agent config and use specified port
- fix(buildless): don't use control port for cache traffic
- docs(buildless): add `docs/Buildless` and root README references

[0]: https://less.build

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 edition = "2021"
 name = "sccache"
 rust-version = "1.67.1"
-version = "0.7.4"
+version = "0.7.5-buildless"
```
Comment (sgammon) [inline Cargo.toml:5]: released under a custom version, for now

File: Cargo.toml [hunk 2]
```diff
@@ -146,10 +146,13 @@ all = [
   "memcached",
   "gcs",
   "azure",
+  "buildless",
   "gha",
   "webdav",
 ]
 azure = ["opendal", "reqsign"]
+buildless = ["opendal", "redis", "webdav"]
+buildless-client = ["buildless", "dist-client", "native-zlib", "vendored-openssl"]
```
Comment (sgammon) [inline Cargo.toml]: `buildless` adds the feature itself, `buildless-client` builds a full target

File: src/config.rs [hunk 3]
```diff
@@ -680,6 +708,40 @@ fn config_from_env() -> Result<EnvConfig> {
         None
     };
 
+    // ======= Buildless =======
+    let buildless = if
+        env::var("SCCACHE_BUILDLESS").is_ok() ||
+        env::var("BUILDLESS_APIKEY").is_ok() {
+        // read kill switch
+        let enable_var = env::var("SCCACHE_BUILDLESS")
```
Comment (sgammon) [inline src/config.rs]: buildless is detected last for configuration, in order to avoid failing tests for other providers which expect to fail if config is incomplete

File: src/cache/buildless.rs [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+        if do_use_agent {
+            agent_config = read_agent_config(Path::new(configpath));
+        } else {
+            agent_config = None;
+        }
+
+        return match transport {
+            BuildlessTransport::AUTO => build_https(do_use_agent, endpoint, apikey, &agent_config),
+            BuildlessTransport::HTTPS => build_https(do_use_agent, endpoint, apikey, &agent_config),
+            BuildlessTransport::RESP => build_resp(do_use_agent, endpoint, apikey),
```
Comment (sgammon) [inline src/cache/buildless.rs:182]: transport selection

File: Cargo.toml [hunk 5]
```diff
@@ -146,10 +146,13 @@ all = [
   "memcached",
   "gcs",
   "azure",
+  "buildless",
   "gha",
   "webdav",
 ]
 azure = ["opendal", "reqsign"]
+buildless = ["opendal", "redis", "webdav"]
```
Comment (sgammon) [inline Cargo.toml:154]: since buildless implements webdav and redis, this code simply reuses those adapters to reduce code surface

File: src/config.rs [hunk 6]
```diff
@@ -704,6 +737,39 @@ fn config_from_env() -> Result<EnvConfig> {
         None
     };
 
+    // ======= Buildless =======
+    let buildless = if env::var("SCCACHE_BUILDLESS").is_ok() || env::var("BUILDLESS_APIKEY").is_ok()
+    {
+        // read kill switch
+        let enable_var = env::var("SCCACHE_BUILDLESS").ok().unwrap_or_default();
```
Comment (sgammon) [inline src/config.rs:741]: buildless takes last precedence over all other adapters (except fallback to local disk), to prevent conflicts with providers which expect to fail with incomplete config.
